File: 012-GUI/cpp/01_read.cpp
#include<opencv2/opencv.hpp>
#include<iostream>

void help(char** argv ) ;

int main(int argc, char* argv[]) { 
	cv::Mat src = cv::imread(argc>1?argv[1]:"./resources/panad.jpg");
    if(src.empty() ) {
        std::cout<<"Failed to open file"<<std::endl;
	    help(argv);
		return -1;
    }
    
    cv::imshow("origin", src);
    cv::waitKey(0);

    cv::destroyWindow("origin");

    return 0;
}

void help(char** argv ) {
	std::cerr<< "\n"
	<< "A simple OpenCV program that loads and displays an image from disk\n"
	<< argv[0] <<" <path/filename>\n"
	<< "For example:" << argv[0] << " ./resources/1.tif\n"
	<< std::endl;
}

File: 012-GUI/python/01_read.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import numpy as np
import cv2

print(cv2.__version__)

img = cv2.imread('resources/panad.jpg')

cv2.imshow("panad", img)

cv2.waitKey(0)

cv2.destroyWindow("panad")
